pycaret/classification.py

```diff
@@ -910,61 +837,15 @@ def setup(
                     "Feature ignored is either target column or doesn't exist in the dataset."
                 )
 
-    # log_experiment
-    if type(log_experiment) is not bool:
-        raise TypeError("log_experiment parameter only accepts True or False.")
-
-    # log_profile
-    if type(log_profile) is not bool:
-        raise TypeError("log_profile parameter only accepts True or False.")
-
     # experiment_name
     if experiment_name is not None:
         if type(experiment_name) is not str:
             raise TypeError("experiment_name parameter must be str if not None.")
 
-    # silent
-    if type(silent) is not bool:
-        raise TypeError("silent parameter only accepts True or False.")
-
-    # remove_perfect_collinearity
-    if type(remove_perfect_collinearity) is not bool:
-        raise TypeError(
-            "remove_perfect_collinearity parameter only accepts True or False."
-        )
-
-    # html
-    if type(html) is not bool:
-        raise TypeError("html parameter only accepts True or False.")
-
     # use_gpu
-    if use_gpu != "Force" and type(use_gpu) is not bool:
+    if use_gpu is not "Force" and not isinstance(use_gpu, bool):
```

Comment:
Identity comparison (is) serves a different purpose than value comparison (==). The latter should be used for strings.